File: examples/sdk_call_site.ts
// Demonstrates prompts extracted from real SDK call sites in TypeScript.
// aiproof walks the TS AST, finds known LLM SDK calls, extracts the string
// arguments (including template literals with ${...} → {0}, {1} placeholders),
// and runs rules against those extracted strings.

import OpenAI from "openai";

const openai = new OpenAI();

export async function askGpt(query: string, name: string): Promise<string> {
  const response = await openai.chat.completions.create({
    model: "gpt-4",
    messages: [
      {
        role: "system",
        content: `You are helping ${name}. Think step by step. Respond with JSON.`,
      },
      {
        role: "user",
        content: `Query: ${query}`,
      },
    ],
    temperature: 0.7,
  });
  return response.choices[0].message.content ?? "";
}

File: examples/sdk_call_site.py
"""Demonstrates prompts extracted from real SDK call sites.

aiproof walks the Python AST, finds known LLM SDK calls, and extracts the
string arguments as Document values. Rules then run against those extracted
strings as if they were standalone prompt files.
"""

from anthropic import Anthropic

client = Anthropic()


def ask_claude(query: str, name: str) -> str:
    response = client.messages.create(
        model="claude-4.7-opus",
        system="Always return exactly this JSON schema. Do not explain.",
        messages=[
            {"role": "user", "content": f"Hello {name}, here is the query: {query}"},
        ],
        temperature=0.9,  # AIP018: deterministic prompt + high temperature is a mismatch
    )
    return response.content[0].text


# LangChain shape — also detected:
from langchain.prompts import PromptTemplate

summarize = PromptTemplate.from_template(
    "Summarize this document: {document}. Output JSON."
)
